http-clients/src/main/java/com/palantir/remoting3/clients/ClientConfigurations.java

```diff
@@ -96,13 +115,22 @@ public static ProxySelector createProxySelector(ProxyConfiguration proxyConfig)
                                 "Expected to find proxy hostAndPort configuration for HTTP proxy")));
                 InetSocketAddress addr = new InetSocketAddress(hostAndPort.getHostText(), hostAndPort.getPort());
                 return fixedProxySelectorFor(new Proxy(Proxy.Type.HTTP, addr));
+            case MESH:
+                return ProxySelector.getDefault(); // MESH proxy is not a Java proxy
             default:
                 // fall through
         }
 
         throw new IllegalStateException("Failed to create ProxySelector for proxy configuration: " + proxyConfig);
     }
 
+    private static Optional<HostAndPort> meshProxy(Optional<ProxyConfiguration> proxy) {
+        if (!proxy.isPresent() || proxy.get().type() != ProxyConfiguration.Type.MESH) {
+            return Optional.empty();
+        }
+        return Optional.of(HostAndPort.fromString(proxy.get().hostAndPort().get()));
```

Comment:
think I'd invert this:
```
if (A && B) {
  return positive;
} else {
  return negative;
}
```

Reply:
but baseline says you should do it the other way!

Reply:
done though :)